okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.java

```diff
@@ -95,10 +95,26 @@ public CallServerInterceptor(boolean forWebSocket) {
         .receivedResponseAtMillis(System.currentTimeMillis())
         .build();
 
+    int code = response.code();
+    if (code == 100) {
+      // server sent a 100-continue even though we did not request one.
+      // try again to read the actual response
+      realChain.eventListener().responseHeadersStart(realChain.call());
```

Comment:
we still have unbalanced calls to responseHeadersStart(). Mind dropping this call too?

Reply:
Oh, right. Fixed in 440531d.